nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/kinesis/stream/ConsumeKinesisStream.java

```diff
@@ -0,0 +1,726 @@
+                        .build()
+                );
+            } else if (dateTimeFormatter != null) {
+                try {
+                    // check the streamTimestamp can be formatted
+                    getStartStreamTimestamp(validationContext, dateTimeFormatter);
+                } catch (Exception e) {
+                    validationResults.add(new ValidationResult.Builder().valid(false)
+                            .subject(STREAM_POSITION_TIMESTAMP.getName())
+                            .explanation(String.format("%s must be parsable by %s", STREAM_POSITION_TIMESTAMP.getDisplayName(),
+                                    TIMESTAMP_FORMAT.getDisplayName()))
+                            .build());
+                }
+            }
+        }
+
+        if (isRecordReaderSet && !isRecordWriterSet) {
+            validationResults.add(new ValidationResult.Builder()
+                    .subject(RECORD_WRITER.getName())
+                    .explanation(String.format("%s must be set if %s is set in order to write FlowFiles as Records.",
+                            RECORD_WRITER.getDisplayName(), RECORD_READER.getDisplayName()))
+                    .valid(false)
+                    .build());
+        } else if (isRecordWriterSet && !isRecordReaderSet) {
+            validationResults.add(new ValidationResult.Builder()
+                    .subject(RECORD_READER.getName())
+                    .explanation(String.format("%s must be set if %s is set in order to write FlowFiles as Records.",
+                            RECORD_READER.getDisplayName(), RECORD_WRITER.getDisplayName()))
+                    .valid(false)
+                    .build());
+        }
+
+        return validationResults;
+    }
+
+    @OnScheduled
+    @Override
+    public void onScheduled(ProcessContext context) {
+        stopped.set(false);
+        super.onScheduled(context);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSessionFactory sessionFactory) {
+        if (worker == null) {
+            synchronized (WORKER_LOCK) {
+                if (worker == null) {
+                    final String workerId = generateWorkerId();
+                    getLogger().info("Starting Kinesis Worker {}", workerId);
+                    // create worker (WorkerState will be CREATED)
+                    worker = prepareWorker(context, sessionFactory, workerId);
+                    // initialise and start Worker (will set WorkerState to INITIALIZING and attempt to start)
+                    new Thread(worker, WORKER_THREAD_NAME_TEMPLATE + workerId).start();
+                }
+            }
+        } else {
+            if (!stopped.get() && WorkerStateChangeListener.WorkerState.SHUT_DOWN == workerState.get()) {
+                throw new ProcessException("Worker has shutdown unexpectedly, possibly due to a configuration issue; check logs for details");
```

Comment:
`context.yield()` needs to be called before throwing the exception (without yielding `onTrigger()` would be called again immediately and would dump the log with the error message in a loop).